compiler/rustc_codegen_ssa/src/back/symbol_export.rs

```diff
@@ -364,14 +394,25 @@ fn exported_symbols_provider_local(
                 MonoItem::Fn(Instance { def: InstanceKind::DropGlue(_, Some(ty)), args }) => {
                     // A little sanity-check
                     assert_eq!(args.non_erasable_generics().next(), Some(GenericArgKind::Type(ty)));
-                    symbols.push((
-                        ExportedSymbol::DropGlue(ty),
-                        SymbolExportInfo {
-                            level: SymbolExportLevel::Rust,
-                            kind: SymbolExportKind::Text,
-                            used: false,
-                        },
-                    ));
+
+                    let root_identifier = match ty.kind() {
+                        rustc_middle::ty::Adt(def, args) => Some((def.did(), args)),
+                        rustc_middle::ty::Closure(id, args) => Some((*id, args)),
+                        _ => None,
+                    };
+                    let should_export = root_identifier
+                        .map_or(true, |(did, args)| is_instantiable_downstream(did, args.types()));
```

Comment:
```suggestion
                    let should_export = match ty.kind() {
                        rustc_middle::ty::Adt(def, args) => is_instantiable_downstream(def.did(), args.types()),
                        rustc_middle::ty::Closure(id, args) => is_instantiable_downstream(*id, args.types()),
                        _ => true,
                    };
```

would be a bit simpler I think.

Reply:
Also I think for drop glue `did` would always be non-local outside of libcore, so we could skip calling `type_of` for it.